### ISSUE TITLE
strictNullCheck does not flag unassigned readonly property

### ISSUE THREAD

**Issue by msterba** [Microsoft/TypeScript#10121]
**TypeScript Version:**  2.0.0 with `--strictNullCheck`

```
class My {
    readonly prop: string;
    dump() {
        console.log(this.prop);
    }
}
```

**Expected behavior:**
Expected a compiler error ('undefined is not assignable' or 'unassigned before use') as this.prop is `undefined`

**Actual behavior:** 
Compiles without complaints and loggs `undefined`.

**Comment by RyanCavanaugh** [comment]
Duplicate of #8476 and #9863 and #9797 and #9773 and #9761 and #9131
